packages/web-client/tests/unit/layer-1-connection-manager-test.ts

```diff
@@ -0,0 +1,204 @@
+    let connectionManager = new ConnectionManager('kovan');
+
+    let onCrossTabConnection = Sinon.spy();
+    connectionManager.on('cross-tab-connection', onCrossTabConnection);
+
+    let message = {
+      type: 'CONNECTED',
+      session: 'SESSION',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onCrossTabConnection.calledWith(message),
+      'The cross-tab-connection callback was called with the sent message'
+    );
+  });
+
+  test('It does not call the cross-tab-connection callback upon connected messages, if already connected', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+    let web3 = {
+      setProvider: () => {},
+    };
+
+    let onCrossTabConnection = Sinon.spy();
+    connectionManager.on('cross-tab-connection', onCrossTabConnection);
+
+    await connectionManager.connect((web3 as unknown) as Web3, 'metamask'); // the test strategy is hardcoded to metamask
+
+    let message = {
+      type: 'CONNECTED',
+      session: 'SESSION',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onCrossTabConnection.notCalled,
+      'The cross-tab-connection callback was not called'
+    );
+  });
+
+  test('It calls the disconnect callback upon disconnected messages, if already connected', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+    let web3 = {
+      setProvider: () => {},
+    };
+
+    let onDisconnect = Sinon.spy();
+    connectionManager.on('disconnected', onDisconnect);
+
+    await connectionManager.connect((web3 as unknown) as Web3, 'metamask'); // the test strategy is hardcoded to metamask
+
+    let message = {
+      type: 'DISCONNECTED',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onDisconnect.calledOnce,
+      'The disconnect callback was called upon receiving the message'
+    );
+  });
+
+  test('It does not call the disconnect callback upon disconnected messages, if not connected', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+
+    let onDisconnect = Sinon.spy();
+    connectionManager.on('disconnected', onDisconnect);
+
+    let message = {
+      type: 'DISCONNECTED',
+    };
+
+    connectionManager.broadcastChannel.test__simulateMessageEvent(message);
+
+    assert.ok(
+      onDisconnect.notCalled,
+      'The disconnect callback was not called upon receiving the message'
+    );
+  });
+
+  test('The onConnect method of the layer 1 connection manager sets the providerId in storage', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+    let web3 = {
+      setProvider: () => {},
+    };
+
+    let onDisconnect = Sinon.spy();
+    connectionManager.on('disconnected', onDisconnect);
```

Comment:
Why do you need to set up a disconnected handler in a test about onConnect?

Reply:
Sorry, I don't. Copy paste error.